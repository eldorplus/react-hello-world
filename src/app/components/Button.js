import React from 'react';

// stateless component
const Button = (props) => {
  return <input type="submit" value={props.text} onClick={props.onClick} />;
};

Button.propTypes = {
  text: React.PropTypes.string,
  onClick: React.PropTypes.func,
};

Button.defaultProps = {
  text: 'Toggle',
  onClick: () => {},
};

export default Button;
