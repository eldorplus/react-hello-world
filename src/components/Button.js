import React from 'react';

// stateless component
const Button = (props) => <input type="submit" value={props.text} onClick={props.onClick} />

Button.propTypes = {
    text: React.PropTypes.string,
};

Button.defaultProps = {
    text: 'Toggle',
};

export default Button;