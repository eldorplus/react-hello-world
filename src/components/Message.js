import React from 'react';

// stateless component
const Message = (props) => {
  return (
    <p>{props.children}</p>
  );
};

Message.propTypes = {
  children: React.PropTypes.node,
};

Message.defaultProps = {
  children: 'Hello World',
};

module.exports = Message;
