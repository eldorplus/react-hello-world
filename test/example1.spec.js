import React from 'react';
import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-addons-test-utils';

import Example1 from '../src/example1';
import Button from '../src/components/Button';
import Message from '../src/components/Message';

const Renderer = ReactTestUtils.createRenderer();

test('Example1 toggles Message on each click', () => {
  const component = renderer.create(
    <Example1 />,
  );
  let tree = component.toJSON();

  tree.children[1].props.onClick();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.children[1].props.onClick();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Example1 renders correctly', () => {
  Renderer.render(<Example1 />);
  const result = Renderer.getRenderOutput();

  expect(result.type).toBe('div');
  expect(result.props.children).toEqual([
    <h2>Example 1 - <small>click once to toggle</small></h2>,
    <Button onClick={result.props.children[1].props.onClick} text="Toggle message" />,
    <Message>Hello world!!!</Message>
  ]);
});