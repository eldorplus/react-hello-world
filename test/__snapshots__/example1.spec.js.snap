exports[`test Example1 toggles Message on each click 1`] = `
<div>
  <h2>
    Example 1 - 
    <small>
      click once to toggle
    </small>
  </h2>
  <input
    onClick={[Function]}
    type="submit"
    value="Toggle message" />
</div>
`;

exports[`test Example1 toggles Message on each click 2`] = `
<div>
  <h2>
    Example 1 - 
    <small>
      click once to toggle
    </small>
  </h2>
  <input
    onClick={[Function]}
    type="submit"
    value="Toggle message" />
  <p>
    Hello world!!!
  </p>
</div>
`;
